Actions/BuildALDoc/BuildALDoc.HelperFunctions.ps1

```diff
@@ -0,0 +1,238 @@
+                }
+            }
+        }
+
+        $arguments = @("init","--output ""$docfxpath""","--loglevel $loglevel","--targetpackages ""$($apps -join '","')""")
+        Write-Host "invoke aldoc $arguments"
+        CmdDo -command $aldocPath -arguments $arguments
+        FixBackslashes -docfxPath $docfxPath
+
+        # Update docfx.json
+        Write-Host "Update docfx.json"
+        $docfxJsonFile = Join-Path $docfxPath 'docfx.json'
+        $docfxJson = Get-Content -Encoding utf8 -Path $docfxJsonFile | ConvertFrom-Json
+        $docfxJson.build.globalMetadata._appName = ReplacePlaceHolders -str $header -version $version -releaseNotes $releaseNotes
+        $docfxJson.build.globalMetadata._appFooter = ReplacePlaceHolders -str $footer -version $version -releaseNotes $releaseNotes
+        $docfxJson | ConvertTo-Json -Depth 99 | Set-Content -Path $docfxJsonFile -Encoding utf8
+
+        Write-Host "docfx.json:"
+        Get-Content $docfxJsonFile | Out-Host
+
+        # Create new toc.yml
+        Write-Host "Create new toc.yml"
+        $tocYmlFile = Join-Path $docfxpath 'toc.yml'
+
+        Write-Host "ORGTOC:"
+        Get-Content $tocYmlFile | Out-Host
+
+        Set-Content -Path $tocYmlFile -Value ($newTocYml -join "`n") -Encoding utf8
+
+        Write-Host "TOC:"
+        Get-Content $tocYmlFile | Out-Host
+
+        $apps | ForEach-Object {
+            $arguments = @("build","--output ""$docfxpath""","--loglevel $loglevel","--source ""$_""")
+            Write-Host "invoke aldoc $arguments"
+            CmdDo -command $aldocPath -arguments $arguments
+            FixBackslashes -docfxPath $docfxPath
+        }
+
+        # Set release notes
+        Write-Host "Update index.md"
+        $indexMdFile = Join-Path $docfxpath 'index.md'
+        Set-Content -path $indexMdFile -value $indexContent -encoding utf8
+
+        Write-Host "index.md:"
+        Get-Content $indexMdFile | Out-Host
+
+
+        $arguments = @("build", "--output ""$docsPath""", "--logLevel $loglevel", """$docfxJsonFile""")
+        if ($hostIt) {
+            $arguments += @('-s')
+            Write-Host "Generate and host site"
+        }
+        Write-Host "invoke doxfx $arguments"
+        CmdDo -command docfx -arguments $arguments
+    }
+    finally {
+        Remove-Item -Path $docfxPath -Recurse -Force
+    }
+}
+
+function CalculateProjectsAndApps {
+    Param(
+        [string] $tempFolder,
+        [string[]] $projects,
+        [string[]] $excludeProjects
+    )
+
+    if ($projects.Count -eq 0) { $projects = @('*') }
+    $projectList = @($projects | ForEach-Object { $_.Replace('\','_').Replace('/','_') })
+    $excludeProjectList = @($excludeProjects | ForEach-Object { $_.Replace('\','_').Replace('/','_') })
+    foreach($mask in 'Apps','Dependencies') {
+        $allApps = @{}
+        Get-ChildItem -Path $tempFolder -Directory | ForEach-Object {
+            if ($_.Name -match "^(.*)-main-$mask-(\d*\.\d*\.\d*\.\d*)$") {
```

Comment:
We do have several places where main is hardcoded and we should never generate documentation from any other branches. There is an item in the plan to enable main branches with a different name, but right now that is not supported.

Reply:
I get that the workflow will most probably be run from _main_ branch, but that doesn't mean that the documentation should be generated for the artifacts only from the _main_ branch.

Reply:
Isn't deploying the doc currently branch-agnostic?

The I read it (might be wrong), but the generated documentation might include the artifacts from releases. Why is it assumed that these releases contain artifacts from the _main_ branch?

Reply:
Correct, currently I generate docs from x GitHub Releases. How you generated these releases doesn't matter, I download the app files from the releases.
But the main documentation (currently limited to main).
I can however see that there is a bug - the deployALDoc might run during CI/CD from other branches as well - that is a bug - should only be the default branch.
We could extend this with multiple branches, but for now - main is the only one
Will fix the bug

Reply:
I have added the hardcoded main to CI/CD so that only main deploys to github_pages.
In the future you might be able to specify which branch to use - but only one branch + releases makes sense - else you might not be able to build the ref documentation (if the build artifacts expired f.ex.)